### PR TITLE
[dbnode] Integrate arrow iterators into tile aggregation

### PR DESCRIPTION
**Pull request by linasm**
**What this PR does / why we need it**:
Uses arrow iterators for large tile aggregation.

**Special notes for your reviewer**:
This is relative to a branch with https://github.com/m3db/m3/pull/2466 and https://github.com/m3db/m3/pull/2451 merged in.

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/dbnode/storage/shard.go [hunk 1]
```diff
@@ -2567,47 +2567,98 @@ func (s *dbShard) AggregateTiles(
...
+					unit, singleUnit := frame.Units().SingleValue()
+					annotation, singleAnnotation := frame.Annotations().SingleValue()
+					if vals := frame.Values(); len(vals) > 0 {
+						lastIdx := len(vals) - 1
+						lastValue := vals[lastIdx]
+						lastTimestamp := frame.Timestamps()[lastIdx]
+						if !singleUnit {
+							// TODO: what happens if unit has changed mid-tile?
+							// Write early and then do the remaining values separately?
+							unit = frame.Units().Values()[lastIdx]
```
Comment (gediminasgu) [inline src/dbnode/storage/shard.go:2626]: This line panics. Many cases when `Values()` returns empty array.

File: src/dbnode/storage/shard.go [hunk 2]
```diff
@@ -2567,47 +2567,98 @@ func (s *dbShard) AggregateTiles(
...
+						}
+						if !singleAnnotation {
+							// TODO: what happens if annotation has changed mid-tile?
+							// Write early and then do the remaining values separately?
+							annotation = frame.Annotations().Values()[lastIdx]
```
Comment (gediminasgu) [inline src/dbnode/storage/shard.go:2631]: Same here. This line panics.

File: src/dbnode/encoding/arrow/tile/recorder.go [hunk 3]
```diff
@@ -63,6 +64,9 @@ func newDatapointRecorder(pool memory.Allocator) *datapointRecorder {
 		nil,
 	)
 
+	//float64s := array.NewFloat64Data(nil)
+	//float64s.Reset(data)
```
Comment (linasm) [inline src/dbnode/encoding/arrow/tile/recorder.go:68]: @arnikola I've commented out this recent addition as it did not compile (and looks like it was a noop).
Might be some unfinished improvement - please have a look.

**Comment by linasm** [comment]
Superseded by https://github.com/m3db/m3/pull/2543

**Pull request closed without merging**